### PR TITLE
Add no-magic-numbers plugin.

### PR DESCRIPTION
**Pull request by trevorgerhardt**
Just a warning for now. Ignores -1, 0, 1 and array indexes. See: http://eslint.org/docs/rules/no-magic-numbers for more.

I'm not 100% sold on this, but it is eye-opening to see how many warnings crop up. I think over the long term this will create better coding practices.

### CODE REVIEW

**Comment by trevorgerhardt** [comment]
This adds ~450 warning to the Analysis UI but only ~15 or so to the DataTools UI. @landonreed is better at defining his variables I guess? :)

**Comment by mattwigway** [comment]
I think this is probably because we deal with time a lot. I've never felt
the need to define 60 seconds in a minute, it's not likely to change.

**Comment by trevorgerhardt** [comment]
@mattwigway agreed. But a variable named `minute` can mean a lot of different things.

File: __tests__/lib/babel.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 /* globals describe, expect, it */
 
+const INTENDED_LENGTH = 2
```
Comment (evansiroky) [inline __tests__/lib/babel.js]: Seems like this should be defined within the test case.  The intended length could be different in other test cases/suites.

File: __tests__/lib/build.js [hunk 2]
```diff
@@ -4,41 +4,40 @@ const build = require('../../lib/build')
...
+      const [cssResults] = build({
         config: {},
         files: [[`${mockDir}/index.css`]]
       })
 
-      const result = await results[0]
+      const result = await cssResults
       const css = result.css
-      expect(css.indexOf('criticalClass')).toBeGreaterThan(-1)
+      expect(css.includes('criticalClass')).toBeTruthy()
```
Comment (evansiroky) [inline __tests__/lib/build.js]: We should replace these with [`expect(...).toContain(item)` syntax](https://facebook.github.io/jest/docs/expect.html#tocontainitem).

File: __tests__/lib/jest.js [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 const jestUtils = require('../../lib/jest')
 
+const EXPECTED_ARGUMENT_LENGTH = 8
```
Comment (evansiroky) [inline __tests__/lib/jest.js]: both of these should probably also be in scope of test case

File: lib/util.js [hunk 4]
```diff
@@ -40,6 +40,7 @@ exports.assertEntriesExist = function (entries) {
 /**
  * A lot of subcommands utilize exact argument lengths. Pop mastarm to handle it.
  */
+const MASTARM_INDEX = 1
 exports.popMastarmFromArgv = function () {
-  process.argv = process.argv.slice(0, 1).concat(process.argv.slice(2))
+  process.argv = process.argv.slice(0, MASTARM_INDEX).concat(process.argv.slice(MASTARM_INDEX + 1))
```
Comment (evansiroky) [inline lib/util.js]: consider using [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) instead

File: __tests__/lib/build.js (hunk 2 above)
Comment (trevorgerhardt) [inline __tests__/lib/build.js]: Ah, didn't realize that existed!

Comment (trevorgerhardt) [inline __tests__/lib/build.js]: The next evolution of "no magic numbers" could be "no magic strings"... 🤔

**Comment by trevorgerhardt** [comment]
Closing for now as I don't think we want this level of strictness in our applications.

**Pull request closed without merging**